### PR TITLE
Add notes for Utf8Char

### PR DESCRIPTION
**Pull request by terrajobst**
@GrabYourPitchforks please make sure I didn't miss anything (spoiler alert: I did because I was writing email towards the end 😈 ).

### CODE REVIEW

File: 2018/System.Text.Utf8Char/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# System.Text.Utf8Char8
```
Comment (jkotas) [inline 2018/System.Text.Utf8Char/README.md]:
```suggestion
# System.Text.Utf8Char
```

File: 2018/System.Text.Utf8Char/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+## JSON Source Package
+
+We discussed several names. Best candidates right now:
+
+* `Microsoft.Bcl.Json.Sources`
+* `Microsoft.Text.Json.Sources`
+
+## Char8
+
+* We need to think about what a down-level port would look like. I don't believe
```
Comment (jkotas) [inline 2018/System.Text.Utf8Char/README.md]: The plan for all *potential* downlevel-ports is best-effort standalone package after it ships in .NET Core in its natural place: https://github.com/dotnet/corefx/issues/32640#issuecomment-427618911

There is not much to think about.

File: 2018/System.Text.Utf8Char/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+[Video](https://www.youtube.com/watch?v=TRKTNxVSzoA)
+
+## JSON Source Package
+
+We discussed several names. Best candidates right now:
+
+* `Microsoft.Bcl.Json.Sources`
+* `Microsoft.Text.Json.Sources`
+
+## Char8
```
Comment (jkotas) [inline 2018/System.Text.Utf8Char/README.md]:
```suggestion
## Utf8Char
```

File: 2018/System.Text.Utf8Char/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+* We need to think about what a down-level port would look like. I don't believe
+  we can rev `System.Memory` package without breaking compat.
+* `ReadOnlySpan<byte>` isn't convenient for UTF8 text
+    - We don't *really* use `ReadOnlySpan<byte>` for text in public APIs. We
+      have a few overloads, but they are mostly in buffer land or on brand new
+      APIs.
+* While there is some validation for individual values, this type doesn't in
+  itself guaranteeing that any `ReadOnlySpan<UtfChar>` is a valid UTF8 string
```
Comment (jkotas) [inline 2018/System.Text.Utf8Char/README.md]:
```suggestion
  itself guaranteeing that any `ReadOnlySpan<Utf8Char>` is a valid UTF8 string
```

File: 2018/System.Text.Utf8Char/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  (for example, it can break multi-byte sequences apart). So if UTF8 text
+  well-formedness is critical, users still have validate. The primary benefit of
+  having the type is to express to callers of an API what the data is supposed
+  to represent. We can also improve debugging by making `ToString()` meaningful.
+* This feature is useful without `Utf8String`. However, the feature is much less
+  compelling without compiler support for UTF8 literals.
+* `IsWellFormedSequence()` doesn't seem super useful. We'd want to perform
```
[the remark this answers is not in the document]
Comment (benaadams) [reply 2018/System.Text.Utf8Char/README.md]: Don't see one, would expect an in-place validate conversion something like (but better named)
```csharp
partial class MemoryExtensions
{
    bool TryValidateAndCast(this Span<byte> data, out Span<Utf8Char> text)
    bool TryValidateAndCast(this ReadOnlySpan<byte> data, out ReadOnlySpan<Utf8Char> text)
}
```

[the remark this answers is not in the document]
Comment (bartonjs) [reply 2018/System.Text.Utf8Char/README.md]: We discussed this in the review, something like

```C#
public partial class Utf8
{
    public static ReadOnlySpan<Utf8Char> ValidateUtf8Bytes(ReadOnlySpan<byte> bytes);
}
```

which would accept bytes, ensure consistency (e.g. `0b11110xxx` is always followed by exactly three `0b10xxxxxx`s), and then `return MemoryMarshal.Cast<byte, Utf8Char>(input);` as an implementation detail.

**Pull request merged**